imap/conversations.c

```diff
@@ -3074,7 +3079,7 @@ EXPORTED const char *conversation_id_encode(int cstate_version,
 
     if (cid == NULLCONVERSATION) {
         buf_setcstr(&buf, "NIL");
-    } else if (cstate_version >= 2) {
+    } else if (USER_COMPACT_EMAILIDS(state)) {
```

Comment:
here it is!  In the case without a cstate below, this used to be true but is now false, because it was `cstate ? cstate->version : 2` which would have returned true, but now it's got a test that requires cstate to be true.

Reply:
I think the old logic was incorrect.  If we don't have cstate, we can't have the use compact ids key set